lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -461,9 +462,11 @@ listTransactions
     :: forall t. (DefineTx t)
     => WalletLayer (SeqState t) t
     -> ApiT WalletId
-    -> Maybe (Iso8601Range "inserted-at")
+    -> Maybe Iso8601Time
+    -> Maybe Iso8601Time
+    -> Maybe SortOrder
     -> Handler [ApiTransaction t]
-listTransactions w (ApiT wid) _maybeRange = do
+listTransactions w (ApiT wid) _maybeStart _maybeEnd _maybeOrder = do
```

Comment:
So… with this scheme, is it possible to have `end < start`?

What would `--start 2019-... --end 2000-...` do depending on `--order ascending` or `--order descending`?

Reply:
> So… with this scheme, is it possible to have `end < start`?
> What would `--start 2019-... --end 2000-...` do depending on `--order ascending` or `--order descending`?

This is an excellent point, and I'm glad you raised it.

We certainly have to clearly define the expected behaviour.

One way to handle this is to treat `start` and `end` as _directionless boundary markers_ when both are specified, and to always sort according to the `order` parameter (with a default of `ascending`, if no order is specified).

So for your example:
* `--start 2019 --end 2000`
      specifies all times between 2000 and 2019 in ascending order.
* `--start 2019 --end 2000 --order ascending`
      specifies all times between 2000 and 2019 in ascending order.
* `--start 2019 --end 2000 --order descending`
      specifies all times between 2000 and 2019 in descending order.

In general, we'd get:

| Specified Start | Specified End | Specified Order | Actual Range | Actual Order |
| - | - | - | - | - |
| _unspecified_ | _unspecified_ | _unspecified_ | all transactions from the start of time until the end of time | ascending |
| _unspecified_ | _unspecified_ | `o` | all transactions from the start of time until the end of time  | `o` |
| `t` | _unspecified_ | _unspecified_ | all transactions from `t` until the end of time | ascending |
| `t` | _unspecified_ | `o` | all transactions from `t` until the end of time | `o` |
| _unspecified_ | `t` | _unspecified_ | all transactions from the start of time until `t` | ascending |
| _unspecified_ | `t` | `o` | all transactions from the start of time until `t` | `o` |
| `t1` | `t2` | _unspecified_ | all transactions between `t1` and `t2` | ascending |
| `t1` | `t2` | `o` | all transactions between `t1` and `t2` | `o` |

What do you think? Does it match your intuition?

Reply:
**Another way**, of course, is to use the **implied** order, if none is specified. So:

So for your example:
* `--start 2019 --end 2000`
      specifies all times between 2000 and 2019 in descending order.
* `--start 2019 --end 2000 --order ascending`
      specifies all times between 2000 and 2019 in ascending order.
* `--start 2019 --end 2000 --order descending`
      specifies all times between 2000 and 2019 in descending order.

Reply:
Whichever way we choose, our documentation and testing should be explicit and consistent.

Reply:
@Anviking I propose that we continue this discussion on the "Implementing filtering on `listTransactions` issue. I've added a comment here: https://github.com/input-output-hk/cardano-wallet/issues/466#issuecomment-515301573

We can implement the actual filtering with a future PR, which will depend on the outcome of the above discussion.

Reply:
@Anviking We've made the decision to throw a `400` error from the API if `start > end`. This will also need appropriate CLI and API tests. In order to keep the code review process simple, I propose tackling this in a further PR. What do you think? I've already added a work item to issue #466.